### ISSUE TITLE
how to connection elasticsearch cluster in docker?

### ISSUE THREAD

**Issue by mswinner** [exceptionless/Exceptionless#1070]
It uses an embedded single node Elasticsearch cluster and does not have backups. It is recommended that you create your own Elasticsearch cluster for production deployments of Exceptionless.

On Linux:

docker run --rm -it -p 5000:80 \
    -v $(pwd)/esdata:/usr/share/elasticsearch/data \
    exceptionless/exceptionless:latest

with my Elasticsearch cluster?? how to write docker  parameter？

thanks!

**Comment by niemyjski** [comment]
You'd need to set the Elasticsearch connection string with an env variable with the key `EX_ConnectionStrings__Elasticsearch` with a value of `server=http://YOUR_SERVER_HERE:9200`. Please note running this image will continue to run elasticsearch in the background even if it's not configured / doing anything.

**Comment by mswinner** [comment]
how to  set Elasticsearch  usename  password   ？

**Comment by mswinner** [comment]
docker run --rm -it -p 5000:80
-e “EX_ConnectionStrings__Elasticsearch: server=http://10.0.0.4:9200”
exceptionless/exceptionless:latest

is this right？

**Comment by niemyjski** [comment]
No, that is not a valid way to set environment variables. The key is `EX_ConnectionStrings__Elasticsearch` and value is `server=http://10.0.0.4:9200` It would be something like this: `EX_ConnectionStrings__Elasticsearch="server=http://10.0.0.4:9200"`. You can find this information in the docker docs. I'd also recommend checking the docs on setting the username/password: https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#access_using_credentials_in_the_url

**Comment by niemyjski** [comment]
Also, since you are using an external Elasticsearch instance you should use the [`exceptionless/app:latest` image](https://hub.docker.com/r/exceptionless/app) instead which just has the api and ui (and not Elasticsearch running inside the image).

**Comment by mswinner** [comment]
docker run --rm -it -p 5000:80
-e “EX_ConnectionStrings__Elasticsearch="server=http://10.0.0.4:9200"”
exceptionless/app:latest

“”.....“”

oh no ！！？

**Comment by niemyjski** [comment]
There is no reason to double quote this. Please refer to the docker docs for more information.
